ddl.dic

```diff
@@ -1076,6 +1096,109 @@ save_enumeration_set.state
 
 save_
 
+save_ENUMERATION_SOURCE
+
+    _definition.id                ENUMERATION_SOURCE
+    _definition.scope             Category
+    _definition.class             Loop
+    _definition.update            2023-06-02
+    _description.text
+;
+    Attributes used to record the original sources of the default values
+    enumerated using the ENUMERATION_DEFAULT category.
+;
+    _name.category_id             ENUMERATION
+    _name.object_id               ENUMERATION_SOURCE
+    _category_key.name            '_enumeration_source.id'
+
+    loop_
+      _description_example.case
+      _description_example.detail
+;
+         loop_
+         _enumeration_source.id
+         _enumeration_source.reference
+         a "International Tables Vol. C Table 4.4.4.1 2nd ed."
+         b "International Tables Vol. H Table 4.3.2.1 1st ed."
+
+         loop_
+         _enumeration_default.index
+         _enumeration_default.value
+         _enumeration_default.source_id
+         H   1  a
+         H-  2  a
+         He  2  b
+         #...
+;
+;
+         There are two sources for the default values given in the enumeration.
+         Entries marked with 'a' come from Table 4.4.4.1 in the 2nd edition of
+         Volume C of the International Tables of Crystallography. Entries marked
+         with 'b' come from Table 4.3.2.1 in the 1st edition of Volume H of the
+         International Tables of Crystallography.
+;
+;
+         _enumeration_source.reference  'doi:10.1107/S2053273322010944 Table S4'
+
+         loop_
+         _enumeration_default.index
+         _enumeration_default.value
+         He  0.8733928
+         Li  1.129319
+         Be  1.592162
+         #...
+;
+;
+         There is a single source for all default values given in the
+         enumeration. All entries can be found in Table S4 of the material
+         referenced by the given DOI.
+;
+
+save_
+
+save_enumeration_source.id
+
+    _definition.id                '_enumeration_source.id'
+    _definition.class             Attribute
+    _definition.update            2023-06-02
+    _description.text
+;
+    A unique identifier of the source material from which the
+    _enumeration_default.value attribute values were taken.
+;
+    _name.category_id             enumeration_source
+    _name.object_id               id
+    _type.purpose                 Key
+    _type.source                  Assigned
+    _type.container               Single
+    _type.contents                Word
+    _method.purpose               Definition
+    _method.expression
+;
+    _enumeration.default = Unique_id(enumeration_source.id)
+;
+
```

Comment:
Why do we need a dREL method here?

Reply:
so that in the event of a single source, you don't need to give an id value; it just gets magicked into existence.

Reply:
The magicking in the case of a single value loop should already happen automatically without the use of dREL code. More specifically, there is an informal agreement that software should be able to imply the value under certain condition, one of which is the one you described (single loop packet with an arbitrary key value).

The use of `Unique_id()` function is reserved for cases when we have to deal with legacy categories for which the id item is newly assigned and thus not present in multiple existing data files (e.g. the `PUBL_AUTHOR` category) category. The use of this function is not suitable here since it also assigns arbitrary values in cases when there is more than one missing key value. This behaviour is not something that we want to allow in newly defined categories since it may lead to certain data anomalies.

Reply:
Roger that.